gcore/geoheif.cpp

```diff
@@ -0,0 +1,225 @@
+/**********************************************************************
+ *
+ * Name:     geoheif.cpp
+ * Project:  GDAL
+ * Purpose:  OGC GeoHEIF shared implementation.
+ * Author:   Brad Hards <bradh@silvereye.tech>
+ *
+ **********************************************************************
+ * Copyright (c) 2024, Brad Hards
+ *
+ * SPDX-License-Identifier: MIT
+ ****************************************************************************/
+#include "geoheif.h"
+
+namespace gdal
+{
+
+//! @cond Doxygen_Suppress
+
+GeoHEIF::GeoHEIF() : gcps(0)
+{
+}
+
+GeoHEIF::~GeoHEIF()
+{
+}
+
+bool GeoHEIF::has_SRS() const
+{
+    return !m_oSRS.IsEmpty();
+}
+
+bool GeoHEIF::has_GCPs() const
+{
+    return haveGCPs;
+}
+
+static double to_double(const uint8_t *data, uint32_t index)
+{
+    uint64_t v = 0;
+    memcpy(&v, data + index, sizeof(v));
+    CPL_MSBPTR64(&v);
+    double d = 0;
+    memcpy(&d, &v, sizeof(d));
+    return d;
```

Comment:
can be further simplified to:
```suggestion
    double d = 0;
    memcpy(&d, data + index, sizeof(d));
    CPL_MSBPTR64(&d);
    return d;
```